### PR TITLE
feat: update cypress icons and add spec icon for .cy files

### LINKED ISSUE #2910: [Icon Request]: Cypress Logo Rebrand
👋🏻 Hello! I'm excited to add the new Cypress logo!
We've rebranded, added a new config file type, and added a new extension to avoid conflicts with Jest/Mocha specs.
https://twitter.com/Cypress_io

- [x] I'm sure this issue is not a _duplicate_.
- [x] I'm submitting an icon request:
  - Type: `extension`, `folder`
  - Icon Name: `cypress`
  - Sample original Icon:  cypress-icon
  - Extensions: `.cy.*` (Just like how `.spec.*` works)
  - Filenames:
    - `cypress.config.{js,ts}`
    - `cypress.json`

### PR DESCRIPTION
**Pull request by elylucas**
Fixes #2910 

**Changes proposed:**

- [X] Add
- [X] Fix

Updates Cypress icons to new branding and adds Cypress spec file icons for *.cy.{js/jsx/ts/tsx} files.

### CODE REVIEW

File: src/iconsManifest/supportedExtensions.ts [hunk 1]
```diff
@@ -1193,8 +1193,22 @@ export const extensions: IFileCollection = {
...
+        `cypress.config.ts`,
+        `cypress.config.mjs`,
+      ],
       filename: true,
+      light: true,
+      format: FileFormat.svg,
+    },
+    {
+      icon: 'cypress_spec',
+      extensions: [`cy.js`, `cy.mjs`, `cy.coffee`, `cy.ts`, `cy.tsx`, `cy.jsx`],
```
Comment (JessicaSachs) [inline src/iconsManifest/supportedExtensions.ts:1209]:
```suggestion
      extensions: [`cy.js`, `cy.mjs`, `cy.cjs`, `cy.coffee`, `cy.ts`, `cy.tsx`, `cy.jsx`],
```

File: src/iconsManifest/supportedExtensions.ts [hunk 2]
```diff
@@ -1193,8 +1193,22 @@ export const extensions: IFileCollection = {
     },
     {
       icon: 'cypress',
-      extensions: [`cypress.json`, `cypress.env.json`],
+      extensions: [
+        `cypress.json`,
+        `cypress.env.json`,
+        `cypress.config.js`,
+        `cypress.config.ts`,
+        `cypress.config.mjs`,
```
Comment (JessicaSachs) [inline src/iconsManifest/supportedExtensions.ts:1201]:
```suggestion
        `cypress.config.cjs`,
        `cypress.config.mjs`,
```

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
Forgot about `.cjs` extensions.

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Minor thing, just add a 2px margin to all the file icons. :)

File: src/iconsManifest/supportedExtensions.ts [hunk 3]
```diff
@@ -1193,8 +1193,23 @@ export const extensions: IFileCollection = {
...
     {
       icon: 'cypress',
-      extensions: [`cypress.json`, `cypress.env.json`],
+      extensions: [
+        `cypress.json`,
+        `cypress.env.json`,
+        `cypress.config.js`,
+        `cypress.config.ts`,
+        `cypress.config.cjs`,
+        `cypress.config.mjs`,
```
Comment (sasial-dev) [inline src/iconsManifest/supportedExtensions.ts:1202]: Really nitpicky, but why are you using template literals here?
```suggestion
        'cypress.json',
        'cypress.env.json',
        'cypress.config.js',
        'cypress.config.ts',
        'cypress.config.cjs',
        'cypress.config.mjs',
```

File: src/iconsManifest/supportedExtensions.ts [hunk 4]
```diff
@@ -1193,8 +1193,23 @@ export const extensions: IFileCollection = {
...
+      ],
       filename: true,
+      light: true,
+      format: FileFormat.svg,
+    },
+    {
+      icon: 'cypress_spec',
+      extensions: [`cy.js`, `cy.mjs`, `cy.cjs`, `cy.coffee`, `cy.ts`, `cy.tsx`, `cy.jsx`],
```
Comment (sasial-dev) [inline src/iconsManifest/supportedExtensions.ts:1210]: Really nitpicky, but why are you using template literals here?
```suggestion
      extensions: ['cy.js', 'cy.mjs', 'cy.cjs', 'cy.coffee', 'cy.ts', 'cy.tsx', 'cy.jsx'],
```

File: src/iconsManifest/supportedExtensions.ts [hunk 5]
```diff
@@ -1193,8 +1193,23 @@ export const extensions: IFileCollection = {
...
+      filename: false,
```
Comment (sasial-dev) [inline src/iconsManifest/supportedExtensions.ts:1211]: This can be omitted.
```suggestion
```

**Review by sasial-dev — CHANGES_REQUESTED** [review]

File: src/iconsManifest/supportedExtensions.ts (hunk 3 above)
Comment (elylucas) [reply src/iconsManifest/supportedExtensions.ts:1202]: wasn't on purpose, thanks

**Review by sasial-dev — DISMISSED** [review]
LGTM!

**Comment by elylucas** [comment]
Hi @sasial-dev, whats the next steps to get this merged in? Thanks!

**Comment by sasial-dev** [comment]
The PR is all ready! This repo (from my experience) leaves PR’s to be merged, and then merges them in batches just before a release is created.

**Review by robertohuertasm — CHANGES_REQUESTED** [review]
Thanks for your contribution. I've reviewed the icons and their size is considerable. Normally we aim for 2Kb.  I've also seen that it seems that these icons have a png inserted. Ideally we should avoid this and just use plain vectors.

**Comment by robertohuertasm** [comment]
I'm preparing a new release for this weekend. Not sure if I will be able to refactor the icons myself as they don't work well with Illustrator and I foresee they will require a little bit of work. If not, we'll release this after this release is completed.

**Comment by elylucas** [comment]
> I'm preparing a new release for this weekend. Not sure if I will be able to refactor the icons myself as they don't work well with Illustrator and I foresee they will require a little bit of work. If not, we'll release this after this release is completed.

Thanks Roberto, I'll see if I can take a look tonight.

**Comment by elylucas** [comment]
@robertohuertasm I optimized the files and took the embedded images out, thanks!

**Comment by robertohuertasm** [comment]
@elylucas thanks for that. I've used illustrator to gain a few more kb from each of the icons. Merging this.

**Review by robertohuertasm — APPROVED** [review]
LGTM 🚀 Thanks for your contribution ❤️

**Pull request merged**